Authenticator/authenticator.py

```diff
@@ -0,0 +1,25 @@
+from jupyterhub.auth import Authenticator
+from tornado import gen
+import csv
+import logging
+
+
+class MyAuthenticator(Authenticator):
+    @gen.coroutine
+    def authenticate(self, data, handler):
+
+        csv_dict={}
+        with open('authentication.csv', mode='r') as csv_file:
+            csv_data = {row["title"]: row["value"] for row in csv.DictReader(csv_file, ("title", "value"))}
+            csv_dict.update(csv_data)
+
+        if data['username'] not in csv_dict:
+            self.logger.warning("No such user: %s", data['username'])
+        else:
+
+            if csv_dict[name] != data['password']:
+                self.logger.Warnig("Incorrect password for user: %s", data['username'])
```

Comment:
This should be `logger.warning`. Try running your Authenticator and exercising these different cases:

```python
authenticator = MyAuthenticator()
authenticator.authenticate({'username': 'missing', 'password': 'match'})
```

to make sure that you cover each case. This will help catch typos.

Reply:
okay, Am on it